### PR TITLE
New provider: AWS ELB Classic

### PR DESCRIPTION
**Pull request by janeczku**
Depends on PR #7

### CODE REVIEW

**Comment by galal-hussein** [comment]
I tested the classic  ELB on Rancher v1.1.4, and it seems to be working perfectly.

**Comment by galal-hussein** [comment]
LGTM

**Comment by deniseschannon** [comment]
https://github.com/rancher/rancher/issues/4659

File: providers/elbv1/aws_elbv1.go [hunk 1]
```diff
@@ -0,0 +1,338 @@
...
+
+	logrus.Debugf("Initialized provider: region: %s, vpc: %s, usePrivateIP %t",
+		p.region, p.vpcID, p.usePrivateIP)
+
+	p.svc, err = elbv1svc.NewService(accessKey, secretKey, p.region, p.vpcID)
+	if err != nil {
+		return err
+	}
+
+	if err := p.svc.CheckAPIConnection(); err != nil {
```
Comment (prachidamle) [inline providers/elbv1/aws_elbv1.go:76]: here you can call p.HealthCheck() that translates to the same p.svc.CheckAPIConnection()? I guess that was the intention to introduce the HealthCheck() (or previous testConnection) in provider interface.

File: providers/elbv1/elbv1svc/elb_classic.go [hunk 2]
```diff
@@ -0,0 +1,362 @@
...
+		}
+	}
+
+	return nil
+}
+
+// EnsureHealthCheckPort ensures that the specified healthcheck
+// uses the specified instance port. If not, it configures it
+// so, while maintaining all it's other properties.
+func (svc *ELBClassicService) EnsureHealthCheckPort(
```
Comment (prachidamle) [inline providers/elbv1/elbv1svc/elb_classic.go:202]: Is this called from the external_lb provider(aws_elbv1.go)? Could not find any references

File: providers/elbv1/elbv1svc/elb_classic.go [hunk 3]
```diff
@@ -0,0 +1,362 @@
...
+	}
+
+	return nil
+}
+
+// EnsureListenerInstancePort ensures that the specified listener
+// is configured to forward traffic to the specified instance port.
+// If not, it recreates the listener and configures it so, while
+// maintaining all it's other properties.
+func (svc *ELBClassicService) EnsureListenerInstancePort(
```
Comment (prachidamle) [inline providers/elbv1/elbv1svc/elb_classic.go:130]: Is this method referred to anywhere?

**Review by prachidamle — APPROVED** [review]
@janeczku Thank you for adding the ELB external LB! Code is LGTM, I have some minor comments that you could address later.

**Pull request merged**